src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -228,11 +350,92 @@ void symbol_factoryt::gen_nondet_init(
                                        : side_effect_expr_nondett(type, loc);
     code_assignt assign(expr, rhs);
     assign.add_source_location()=loc;
-
+    if(expr.id() == ID_symbol)
+    {
+      auto const &symbol_expr = to_symbol_expr(expr);
+      auto const associated_array_size =
+        get_deferred_size(get_symbol_base_name(symbol_expr));
+      if(associated_array_size.has_value())
+      {
+        assign.rhs() = typecast_exprt{
+          symbol_table.lookup_ref(associated_array_size.value()).symbol_expr(),
```

Comment:
Hm, does this work? I thought `lookup_ref()` requires the full name of a symbol whereas `associated_array_size` is the base name I think.

Reply:
Ah, this does work but is confusingly worded. The `associated_array_size` here is actually the _deferred_ array size symbol which we dynamically created earlier, and we do have the full name for that. I'll see if I can make this clearer